plugins/out_cloudwatch_logs/cloudwatch_api.c

```diff
@@ -520,12 +522,160 @@ int add_event(struct flb_cloudwatch *ctx, struct cw_flush *buf,
     return 0;
 }
 
+int should_add_to_emf(struct flb_intermediate_metric *an_item)
+{
+    /* Valid for cpu plugin */
+    if (strncmp(an_item->key.via.str.ptr, "cpu_p", 4) == 0 
+    || strncmp(an_item->key.via.str.ptr, "user_p", 6) == 0 
+    || strncmp(an_item->key.via.str.ptr, "system_p", 8) == 0) {
+        return 1;
+    }
+
+    /* Valid for mem plugin */
+    if (strncmp(an_item->key.via.str.ptr, "Mem.total", 9) == 0 
+    || strncmp(an_item->key.via.str.ptr, "Mem.used", 8) == 0 
+    || strncmp(an_item->key.via.str.ptr, "Mem.free", 8) == 0 
+    || strncmp(an_item->key.via.str.ptr, "Swap.total", 10) == 0 
+    || strncmp(an_item->key.via.str.ptr, "Swap.used", 9) == 0 
+    || strncmp(an_item->key.via.str.ptr, "Swap.free", 9) == 0) {
+        return 1;
+    }
+
+    return 0;
+}
+
+struct msgpack_object pack_emf_payload(struct flb_cloudwatch *ctx, 
+                                       struct mk_list *flb_intermediate_metrics, 
+                                       const char *input_plugin, 
+                                       struct flb_time tms)
+{
+    struct mk_list *metric_temp;
+    struct mk_list *metric_head;
+
+    /* msgpack::sbuffer is a simple buffer implementation. */
+    msgpack_sbuffer sbuf_emf;
+    msgpack_sbuffer_init(&sbuf_emf);
+
+    /* Serialize values into the buffer using msgpack_sbuffer_write */
+    msgpack_packer packer_emf;
```

Comment:
packer name must be ```mp_pck``` and sbuffer ```mp_sbuf``` (we use the same name in core and plugins)

Reply:
Updated.